docs/pages/router/reference/authentication.mdx

```diff
@@ -5,115 +5,352 @@ description: How to protect routes with Expo Router.
+```jsx app/(app)/index.js
 import { Text, View } from 'react-native';
 
-import { useAuth } from '../context/auth';
+import { useSession } from '../../ctx';
 
 export default function Index() {
-  const { signOut } = useAuth();
+  const { signOut } = useSession();
   return (
     <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
-      <Text onPress={() => signOut()}>Sign Out</Text>
+      <Text
+        onPress={() => {
+          // The `app/(app)/_layout.tsx` will redirect to the sign-in screen.
+          signOut();
+        }}>
+        Sign Out
+      </Text>
     </View>
   );
 }
 ```
 
-Now if the authentication state changes globally, the user will be redirected to the appropriate route.
+We now have an app that can present a loading state while it checks the initial authentication state, and redirect to the sign-in screen if the user is not authenticated. If a user deep links to any protect routes, they'll be redirect back to the sign-in screen.
+
+## Alternative loading states
+
+In Expo Router, something must be rendered to the screen during the loading of the initial auth state. In the example above, we render a loading state in the app layout. Alternatively, you can make the `index` route a loading state, and move the initial route to something like `/home`, this is akin to how Twitter works.
+
+## Modals and per-route authentication
+
+Another common pattern is to render a sign-in modal over the top of the app. This enables you to dismiss partially preserve deep links when the authentication is complete. However, this pattern requires routes to be rendered in the background, these routes will need to handle data loading without authentication.
+
+<FileTree
+  files={[
+    ['app/_layout.js', 'Declares global session context'],
+    'app/(app)/_layout.js',
+    ['app/(app)/sign-in.js', <span>Modally presented over the root</span>],
+    ['app/(app)/(root)/_layout.js', <span>Protects child routes</span>],
+    [
+      'app/(app)/(root)/index.js',
+      <span>
+        Requires authorization <Lock01Icon className="inline mb-1" />
+      </span>,
+    ],
+  ]}
+/>
+
+```jsx app/(app)/_layout.js
+import { Stack } from 'expo-router';
+
+export const unstable_settings = {
+  initialRouteName: '(root)',
+};
+
+export default function AppLayout() {
+  return (
+    <Stack>
+      <Stack name="(root)" />
+      <Stack
+        name="sign-in"
+        options={{
+          presentation: 'modal',
+        }}
+      />
+    </Stack>
+  );
+}
+```
+
+## Navigating without navigation
+
+If you encounter the following error:
+
+```
+Error: Attempted to navigate before mounting the Root Layout component. Ensure the Root Layout component is rendering a Slot, or other navigator on the first render.
+```
+
+This can occur when the app attempts to perform navigation without a navigator mounted in the [root layout](/router/advanced/root-layout). You can fix this by adding a group and moving conditional logic up a level.
```

Comment:
```suggestion
This can occur when the app attempts to perform navigation without a navigator mounted in the [root layout](/router/advanced/root-layout). You can fix this by adding a group and moving conditional logic down a level.
```

Wouldn't this be moving down a level since the logic is now deeper in the file hierarchy and in a nested component?